action/action.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	actions "github.com/sethvargo/go-githubactions"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+var defaultRepositoryOwner string = "hashicorp"
+var defaultMetadataFileName string = "metadata.json"
+
+type args struct {
+	name  string
+	value string
+}
+
+const (
+	repository       = iota
+	repositoryOwner  = iota
+	filePath         = iota
+	metadataFileName = iota
+	version          = iota
+)
+
+type Metadata struct {
+	Product         string `json:"repository"`
+	Org             string `json:"repositoryOwner"`
+	Revision        string `json:"sha"`
+	Version         string `json:"version"`
+	BuildWorkflowId string `json:"buildWorkflowId"`
```

Comment:
There are additional values we need set now (branch and repo) https://github.com/hashicorp/consul-terraform-sync/blob/747534cb447522ed4cfcb835ccc750c0a651440a/.github/scripts/generate_metadata.sh#L25 (it doesn't look like these have been updated in the helloworld repo yet so it prob wasn't obvious!). I think it would be fine here to just grab these values using the GITHUB env vars like you have done with the GITHUB_RUN_ID. 
Similarly I think we can get the sha/revision and org the same way using GITHUB_SHA and (part of) GITHUB_REPOSITORY rather than having to exec out to get the sha or pass in the org or set a default value for it.

Reply:
thank you for catching this, fixing it

Reply:
done added extra fields